rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml

```diff
@@ -130,3 +130,37 @@ setup:
   - is_true: metadata
   - is_true: routing_table
   - is_true: routing_nodes
+
+---
+"Filtering the cluster state by indices using wildcards should work in routing table and metadata":
+  - do:
+      index:
+        index: index1
+        type:  type
+        id:    testing_document
+        body:
+            "text" : "The quick brown fox is brown."
+
+  - do:
+      index:
+        index: index2
+        type:  type
+        id:    testing_document
+        body:
+            "text" : "The quick brown fox is brown."
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      cluster.state:
+        metric: [ routing_table, metadata ]
+        index: [ index* ]
+
+  - is_false: metadata.indices.testidx
+  - is_false: routing_table.indices.testidx
+
+  - is_true: metadata.indices.index1
+  - is_true: routing_table.indices.index1
+  - is_true: metadata.indices.index2
+  - is_true: routing_table.indices.index2
```

Comment:
I cannot help but wonder if we should keep on adding these types of tests as REST tests. Are they really useful to language clients? Or is it just the only way that we had up until now to test the REST layer? I don't mean to block this specific change, just wondering what we should do in the future. Would be nice to unit test this stuff more, or maybe use ESRestTestCase that @nik9000 recently introduced, especially with cat apis that only return text, so no json parsing involved.